### PR TITLE
ECL-173: Added AML supervisor selection

### PR DESCRIPTION
**Pull request by TonyCOgilvie**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+) {
+
+    <h1 class="govuk-heading-l">@messages("notLiable.heading")</h1>
+
+    <p class="govuk-body">@messages("notLiable.body", EclTaxYear.currentFinancialYear)</p>
+
+    <div class="govuk-inset-text">
+        <p>@messages("notLiable.inset.p1")</p>
+        <p>@Html(messages("notLiable.inset.p2", messages("notLiable.inset.link")))</p>
+    </div>
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html:36]: There is a twirl component for inset text we should use, see: https://design-system.service.gov.uk/components/inset-text/#injected-example-play

File: conf/messages.en [hunk 2]
```diff
@@ -58,6 +58,18 @@ ukRevenue.heading = What was your UK revenue in {0}?
...
+
+notLiable.title = You do not need to register
+notLiable.heading = You do not need to register for the Economic Crime Levy
+notLiable.body = You do not need to register to pay the Economic Crime Levy (ECL) because your UK revenue for {0} does not exceed £10.2 million.
+notLiable.inset.p1 = You will need to register for the ECL in the future if your UK revenue equals or exceeds £10.2 million.
+notLiable.inset.p2 = The {0} has more information on who needs to pay the levy.
+notLiable.inset.link = <a href="https://www.gov.uk/government/publications/economic-crime-anti-money-laundering-levy/economic-crime-anti-money-laundering-levy" class="govuk-link">Economic Crime Levy guidance</a>
+notLiable.feedback.heading = Tell us what you think
+notLiable.feedback.p1 = This is a new service and we want to know what you think of it.
+notLiable.feedback.link = <a href="#" class="govuk-link">Send us your feedback</a>
```
Comment (peter-hazell) [inline conf/messages.en:71]: you could parameterise the href link here as `{0}` and pass through the exit survey URL from the app config

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala [hunk 3]
```diff
@@ -47,6 +47,20 @@ class UkRevenueController @Inject() (
...
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
+        meetsRevenueThreshold =>
+          eclRegistrationConnector
+            .upsertRegistration(request.registration.copy(meetsRevenueThreshold = Some(meetsRevenueThreshold)))
+            .flatMap { _ =>
+              if (meetsRevenueThreshold) {
+                Future.successful(Ok("Proceed"))
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala:59]: I assume we are planning on implement the next page as part of this as well?

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala [hunk 4]
```diff
@@ -47,6 +47,20 @@ class UkRevenueController @Inject() (
...
+              } else {
+                Future.successful(Redirect("you-do-not-need-to-register"))
```
[the remark this answers is not in the document]
Comment (TonyCOgilvie) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala]: 👍

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html (hunk 1 above)
Comment (TonyCOgilvie) [reply app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html]: 👍

File: conf/messages.en (hunk 2 above)
Comment (TonyCOgilvie) [reply conf/messages.en]: 👍

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala (hunk 3 above)
Comment (TonyCOgilvie) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala:60]: Yes, once you are happy with what I've done so far I will move onto this page.

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ *@
+
+@import uk.gov.hmrc.economiccrimelevyregistration.utils.EclTaxYear
+
+@this(
+        layout: templates.Layout,
+        govukInsetText : GovukInsetText
+)
+
+@(feedbackUrl: String)(implicit request: Request[_], messages: Messages)
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/views/NotLiableView.scala.html:24]: you don't need to pass `feedbackUrl` as a parameter into the view, you can inject `AppConfig` into your twirl templates as part of the `@this(...`

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/NotLiableController.scala [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class NotLiableController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  view: NotLiableView
+) extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = Action { implicit request =>
```
Comment (TonyCOgilvie) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/NotLiableController.scala:33]: Requires authorise action

File: conf/messages.en [hunk 7]
```diff
@@ -58,13 +58,49 @@ ukRevenue.heading = What was your UK revenue in {0}?
...
 
-checkYourAnswers.title = Check your answers
-checkYourAnswers.heading = Check your answers
+notLiable.title = You do not need to register
+notLiable.heading = You do not need to register for the Economic Crime Levy
+notLiable.body = You do not need to register to pay the Economic Crime Levy (ECL) because your UK revenue for {0} does not exceed £10.2 million.
+notLiable.inset.p1 = You will need to register for the ECL in the future if your UK revenue equals or exceeds £10.2 million.
+notLiable.inset.p2 = The {0} has more information on who needs to pay the levy.
+notLiable.inset.link = <a href="https://www.gov.uk/government/publications/economic-crime-anti-money-laundering-levy/economic-crime-anti-money-laundering-levy" class="govuk-link">Economic Crime Levy guidance</a>
+notLiable.feedback.heading = Tell us what you think
```
Comment (TonyCOgilvie) [inline conf/messages.en:69]: If this feedback text and link is to be used elsewhere consider making the keys generic and move them out of this section.

File: app/uk/gov/hmrc/economiccrimelevyregistration/models/AmlSupervisorType.scala [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+            value = Some(value.toString),
+            id = Some(s"value_$index"),
+            conditionalHtml = Some(selectAmlProfessionalBody)
+          )
+        case _     =>
+          RadioItem(
+            content = Text(messages(s"amlSupervisor.${value.toString}")),
+            value = Some(value.toString),
+            id = Some(s"value_$index")
+          )
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/models/AmlSupervisorType.scala:91]: think we can shorten this by just putting a if statement where we assign the value to `conditionalHtml` e.g.

`if(value == Other) Some(selectAmlProfessionalBody) else None`

File: app/uk/gov/hmrc/economiccrimelevyregistration/models/Registration.scala [hunk 9]
```diff
@@ -23,11 +23,22 @@ final case class Registration(
...
 )
 
 object Registration {
   implicit val format: OFormat[Registration] = Json.format[Registration]
+
+  def empty(internalId: String): Registration = Registration(
+    internalId = internalId,
+    entityType = None,
+    meetsRevenueThreshold = None,
+    amlSupervisor = None,
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/models/Registration.scala:39]: reminder to update the backend model 😄

Comment (TonyCOgilvie) [reply app/uk/gov/hmrc/economiccrimelevyregistration/models/Registration.scala:39]: Updated Registration model on backend microservice - see separate PR

**Review by peter-hazell — APPROVED** [review]

**Pull request merged**